src/features/core/metrics/errorReporting/providers/Bugsnag.js

```diff
@@ -0,0 +1,240 @@
+];
+const BUGSNAG_ERROR_FILE_ORIGIN_KEY = 'stacktrace.0.file';
+
+const getReleaseStage = () => {
+  const host = window.location.hostname;
+  const devHosts = ['www.rs-unit-test-host.com', 'localhost', '127.0.0.1', '[::1]'];
+
+  return host && devHosts.includes(host) ? 'development' : '__RS_BUGSNAG_RELEASE_STAGE__';
+};
+
+const isValidVersion = (globalLibInstance) => {
+  // For version 7
+  let version =
+    globalLibInstance &&
+    globalLibInstance._client &&
+    globalLibInstance._client._notifier &&
+    globalLibInstance._client._notifier.version;
+
+  // For versions older than 7
+  if (!version) {
+    let tempInstance = globalLibInstance({
+      apiKey: API_KEY,
+      releaseStage: 'version-test',
+      // eslint-disable-next-line func-names, object-shorthand
+      beforeSend: function () {
+        return false;
+      },
+    });
+    version = tempInstance.notifier && tempInstance.notifier.version;
+    tempInstance = undefined;
+  }
+
+  return version && version.charAt(0) === BUGSNAG_VALID_MAJOR_VERSION;
+};
+
+const isRudderSDKError = (event) => {
+  const errorOrigin = get(event, BUGSNAG_ERROR_FILE_ORIGIN_KEY);
+
+  if (!errorOrigin || typeof errorOrigin !== 'string') {
+    return false;
+  }
+
+  const srcFileName = errorOrigin.substring(errorOrigin.lastIndexOf('/') + 1);
+  return SDK_FILE_NAMES.includes(srcFileName);
+};
+
+const enhanceErrorEventMutator = (event, metadataSource) => {
+  event.updateMetaData('source', {
+    metadataSource,
+  });
+
+  const { errorMessage } = event;
+  // eslint-disable-next-line no-param-reassign
+  event.context = errorMessage;
+
+  // Hack for easily grouping the script load errors
+  // on the dashboard
+  if (errorMessage.includes('error in script loading')) {
+    // eslint-disable-next-line no-param-reassign
+    event.context = 'Script load failures';
+  }
+
+  // eslint-disable-next-line no-param-reassign
+  event.severity = 'error';
+};
+
+const loadBugsnagSDKScript = (name) => {
+  const isNotLoaded = GLOBAL_LIBRARY_OBJECT_NAMES.every(
+    (globalKeyName) => !Object.hasOwn(window, globalKeyName),
+  );
+
+  if (isNotLoaded) {
+    ScriptLoader(name, BUGSNAG_CDN_URL, {
+      isNonNativeSDK: 'true',
+      async: 'false',
+      skipDatasetAttributes: true,
+    });
+  }
+};
+
+class BugsnagProvider {
+  constructor(sourceId, onClientReady) {
+    this.pluginName = ERROR_REPORT_PROVIDER_NAME_BUGSNAG;
+    this.sourceId = sourceId;
+    this.onClientReady = onClientReady;
+    this.initClientOnLibReadyInterval = undefined;
+    this.init();
+  }
+
+  /**
+   * The responsibility of this function is to check Bugsnag native SDK
+   * has been loaded or not in a certain interval.
+   * If already loaded initialize the SDK.
+   */
+  init() {
+    // Return if RS Bugsnag instance is already initialized
+    if (window.analytics && window.analytics[ERROR_REPORTING_SERVICE_GLOBAL_KEY_NAME]) {
```

Comment:
This should be `window.rudderanalytics`, right?

Reply:
done